spring-core/src/main/java/org/springframework/util/StringUtils.java

```diff
@@ -810,8 +811,9 @@ public static TimeZone parseTimeZoneString(String timeZoneString) {
 		}
 		List<String> result = new ArrayList<>();
 		result.addAll(Arrays.asList(array1));
+		Set<String> stringSet = new HashSet<>(result);
```

Comment:
yes, i did mistake) i can say that it would be really faster on large string arrays, but slower on small arrays.